pydantic/types.py

```diff
@@ -2800,3 +2804,21 @@ class Model(BaseModel):
             _AllowAnyJson,
         ],
     )
+
+
+class _OnErrorOmit:
```

Comment:
I wonder if we should we make this a dataclass?

```py
@dataclass
class OnError(Generic[T]):
    on_error: Literal['omit', 'raise', 'default']
    default: Any | Callable[[], T] = None
```
?

Reply:
It sounds like you want to fully expose the `with_default_value` options, but IMO even if we do that later on this use case is simple and common enough to warrant it's own annotation with no options so you don't need to do `Annotated[..., OnError(on_error='omit')]`.

Reply:
PR welcome for anyone to add `OnError` as described by me above.